### ISSUE TITLE
UI: Freeze when deleting record 

### ISSUE THREAD

**Issue by tevintchuinkam** [pocketbase/pocketbase#3299]
Hey, 

I sometimes run into the problem where I cannot delete a record. It just freezes.  I see no apparent reason why the delete would fail. 

![image](https://github.com/pocketbase/pocketbase/assets/67971452/b771cb59-9943-407f-b5ec-90a12c7db7a9)

Network Tab in Chrome shows this pending request:
![image](https://github.com/pocketbase/pocketbase/assets/67971452/becb0c53-a387-4c77-a357-e66bd961f9b6)

I am using github.com/pocketbase/pocketbase v0.18.4 

No error logs appear while the pocketbase server is still running, but as soon as I stop it i get this error here: 

![image](https://github.com/pocketbase/pocketbase/assets/67971452/d02ab871-3dc1-47ef-af6c-b1e2b100b3de)

**Comment by ganigeorgiev** [comment]
Do you have "Cascade delete" enabled? If yes, then the delete operation could be slow in case you have a lot of related records to delete/update.

The above logs also looks strange. Do you have custom Go code that attach to the delete hooks?

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it. Just tested it with deleting 100+ records (30+ individual delete requests with 2 nested cascade delete) and it executes ~1-2s.

If you are still not able to identify the issue, please create a minimal reproducible repo with sample data and I'll have a look at it.

**Comment by ganigeorgiev** [comment]
_I also forgot to mention that another thing that can also block the delete operation (or any other write) is if there is a running backup operation because it is executed in a transaction._

**Comment by tevintchuinkam** [comment]
> I'm not able to reproduce it. Just tested it with deleting 100+ records (30+ individual delete requests with 2 nested cascade delete) and it executes ~1-2s.
> 
> If you are still not able to identify the issue, please create a minimal reproducible repo with sample data and I'll have a look at it.

I created a minimal example to showcase my problem. It's available under [this repository](https://github.com/tevintchuinkam/pocketbase-minimal-example).

I added the migrations folder, and a setup function so you should have the right environment as soon as you start the app. 
The problem arises when attempting to delete either the "message" or the "conversation" or the "book". 

Would love if you could take a look at it, 

Thank you

**Comment by ganigeorgiev** [comment]
I haven't run the repo but I think the problem is the same as https://github.com/pocketbase/pocketbase/discussions/3205 and it is caused by the `OnModelBeforeDelete` hook in https://github.com/tevintchuinkam/pocketbase-minimal-example/blob/main/main.go#L69.

The reason for the hang is that when deleting a record it starts a transaction that will prevent other writes (aka. the `SaveRecord` calls).

To avoid it, you can use the event hook `Dao` to ensure that the `SaveRecord` operations are executed within the same transaction.

In other words - inside the hook, try to replace `app.Dao()` with `e.Dao`.

**Comment by tevintchuinkam** [comment]
Yep, that was it. Many thanks!
